recipe/build-mpi.sh

```diff
@@ -73,6 +73,10 @@ fi
 make -j"${CPU_COUNT:-1}"
 make install
 
+# openmpi installs .mod files in the wrong prefix (/lib instead of /include)
+ls -l $PREFIX/lib/*.mod
+mv -v $PREFIX/lib/*.mod $PREFIX/include/
```

Comment:
What about fixing the include  paths?

```
$ grep 'compiler_flags=' share/openmpi/mpif*-wrapper-data.txt 
share/openmpi/mpif77-wrapper-data.txt:compiler_flags=-I${includedir}   -I${libdir}
share/openmpi/mpif90-wrapper-data.txt:compiler_flags=-I${includedir}   -I${libdir}
share/openmpi/mpifort-wrapper-data.txt:compiler_flags=-I${includedir}   -I${libdir}
```

Reply:
Using `mv` could this have backward compatibility implications. Maybe you should keep symlinks in `libdir` ?

Reply:
I think it's quite unlikely, since mpifort puts both $PREFIX/include and $PREFIX/lib on the include path (with OMPI_FCFLAGS, it's only $PREFIX/include) but we can copy to be sure.

Reply:
used symlinks

Reply:
I was thinking about users setting flags "by hand", that is, not using MPI compiler wrappers.

Reply:
I don't think they can, because all the `.h` headers like `mpif.h` are in include, so openmpi needs _both_ `-I$PREFIX/include -I$PREFIX/lib`. I'm not sure if there are any feasible situations where:

1. `mpifort` isn't used (not affected)
2. `-I$PREFIX/lib` is given, and
3. `-I$PREFIX/include is _not_ given (i.e. no 'normal' headers or modules are required)

all of which must be true for moving to break anything.

Reply:
Those using `use mpi` or `use mpi_f08` modules do not need `mpif.h` but just the `mpi_*.mod` files. Therefore it could be possible that some folks where using  a bare `gfortran -I$PREFIX/lib -L$PREFIX/lib ...` and things would just work, but then things would break after your changes if you had not kept symlinks in libdir.

Reply:
Yeah, I guess it's technically not impossible, though they would have to use mpi, nothing else, and only openmpi.